frontend/packages/pipelines-plugin/src/components/pipelines/detail-page-tabs/pipeline-details/PipelineVisualizationTask.tsx

```diff
@@ -100,6 +100,22 @@ export const PipelineVisualizationTask: React.FC<PipelineVisualizationTaskProp>
   }
 
   let resources;
+  if (task.taskSpec) {
+    return (
+      <TaskComponent
+        pipelineRunName={pipelineRunName}
+        task={{ data: { spec: task.taskSpec } }}
+        name={task.name || ''}
+        namespace={namespace}
+        status={taskStatus}
+        isPipelineRun={!!pipelineRunStatus}
+        disableTooltip={disableTooltip}
+        selected={selected}
+        width={width}
+        height={height}
+      />
+    );
+  }
```

Comment:
PipelineVisualizationTask accepts `disableTooltip` and if that is passed, then the control will early return and it will not reach here at all. I think this will be problem when the edit pipeline option is enabled. Better return this one first based on taskSpec's availability.